src/core/common/encoding.hpp

```diff
@@ -99,6 +99,63 @@ inline uint64_t HostSwap64(uint64_t v) { return Swap64(v); }
 
 #endif  // LITTLE_ENDIAN
 
+/**
+ * This function reads a `uint16_t` value from a given buffer assuming big-ending encoding.
+ *
+ * @param[in] aBuffer   Pointer to buffer to read from.
+ *
+ * @retval The `uint16_t` value read from buffer.
```

Comment:
nit: `@retval` is typically used to document specific return values (e.g. error codes), use `@returns` instead.  More instances below.